Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -527,7 +528,12 @@ public ParserState GetOrCreateModuleState(IVBComponent component)
 
         public ParserState GetModuleState(IVBComponent component)
         {
-            return _moduleStates.GetOrAdd(new QualifiedModuleName(component), new ModuleState(ParserState.Pending)).State;
+            return GetModuleState(new QualifiedModuleName(component));
+        }
+
+        public ParserState GetModuleState(QualifiedModuleName module)
+        {
+            return _moduleStates.GetOrAdd(module, new ModuleState(ParserState.Pending)).State;  //Why does getting the module state potentially create a new one? That is not too intuitive.
```

Comment:
I will remove the comment.
The thing I find unintuitive about the implementation is that this _function_ has a side-effect if the module does not exist. I would actually expect that it throws an exeption at me whenever I want to know the state of something that is not there, in particular if there is another method called `GetOrCreateModuleState`.